pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -1040,6 +1040,24 @@ def create_split(self, source_id: NodeIdLike, splits: Iterable[tuple[NodeIdLike,
         self.add_successor(split)
         return split
 
+    def create_split_by_nodes(self, source_id: NodeIdLike, target_ids: Iterable[NodeIdLike]) -> KCFG.Split | None:
+        """Create a split without crafting a CSubst."""
+        source = self.node(source_id)
+        targets = [self.node(nid) for nid in target_ids]
+        csubsts: list[CSubst] = []
+
+        for target in targets:
+            subst = source.cterm.config.match(target.cterm.config)
+            if subst is None:
+                return None
+            source_constraints = [subst(c) for c in source.cterm.constraints]
+            constraints = [c for c in target.cterm.constraints if c not in source_constraints] + [
+                c for c in source_constraints if c not in target.cterm.constraints
+            ]
+            csubsts.append(CSubst(subst, constraints))
+
+        return self.create_split(source.id, zip(target_ids, csubsts, strict=True))
```

Comment:
Create a split edge from source to targets. This won't modify the nodes of source and targets. So users need to ensure that source and targets conform to the implication relationship. The generated CSubst contains the delta constraints between source and targets. Therefore, CSubst.apply(source) may not be equal to target, but it can definitely obtain a target that conforms to the implication relationship.